content/en/docs/concepts/workloads/controllers/daemonset.md

```diff
@@ -201,6 +201,39 @@ them according to its `updateStrategy`.
 
 You can [perform a rolling update](/docs/tasks/manage-daemon/update-daemon-set/) on a DaemonSet.
 
+While working with the rolling updates of the DaemonSets, you will come across two important parameters: `maxSurge` and `maxUnavailable`.
+
+### MaxSurge 
+
+This parameter is responsible for determining the maximum number of pods that can be created above the desired number of pods mentioned in the DaemonSet.
+While performing a rolling update, Kubernetes will keep creating new pods to replace the old ones. 
+`maxSurge` helps you to regulate the surge of additional pods beyond the desired count. 
+Let's look at an example below: 
+Suppose the value of `maxSurge` is set to 1 and the desired count is 5, Kubernetes can create up to 6 pods while performing the update process.
+This parameter is useful when you want to temporarily upscale the number of pods while performing the update.
+
+### MaxUnavailable
+
+This particular parameter defines the maximum number of pods that can be unavailable while performing the update.
+In simpler terms, they are being deleted and not yet replaced by new pods, during the update process. 
+`maxUnavailable` can specify the number or percentage of pods that can be available at that particular moment. Let's look at an example below: 
+
+Suppose the value of `maxUnavailable` is set to 1 and the desired count is 5,
+Kubernetes can delete and replace pods one at a time, ensuring the fact that there should always be at least 4 pods available.
+This parameter is useful to maintain the desired level of availability during updates.
+
+You need to set `.spec.updateStrategy.rollingUpdate.maxUnavailable` (default to 1), `.spec.minReadySeconds` (default to 0) and `.spec.updateStrategy.rollingUpdate.maxSurge` (defaults to 0).
+Please note that the default value of `maxSurge` and `maxUnavailable` is 25% of `replicas`.
```

Comment:
Hi @sftim @tanayvaswani [here](https://github.com/kubernetes/kubernetes/blob/af99df6a6bead927d61746dc431fc1b526af55a0/pkg/apis/apps/types.go#L497) the code file says: **By default, a value of 1 is used.**
```go
// The maximum number of pods that can be scheduled above the original number of
	// pods.
	// Value can be an absolute number (ex: 5) or a percentage of total pods at
	// the start of the update (ex: 10%). This can not be 0 if MaxUnavailable is 0.
	// Absolute number is calculated from percentage by rounding up.
	// By default, a value of 1 is used.
	// Example: when this is set to 30%, the new RC can be scaled up by 30%
	// immediately when the rolling update starts. Once old pods have been killed,
	// new RC can be scaled up further, ensuring that total number of pods running
	// at any time during the update is at most 130% of original pods.
	// +optional
	MaxSurge intstr.IntOrString
``` 
Is this stated right?

Reply:
I tested on my system for deployment, and the `kubectl get deployment nginx-deployment -oyaml` command shows maxSurge and maxUnavailable to be 25% when we don't state them explicitly.
for more reference : https://github.com/kubernetes/kubernetes/blob/af99df6a6bead927d61746dc431fc1b526af55a0/pkg/apis/apps/v1/defaults.go#L59

```go
	if strategy.RollingUpdate.MaxSurge == nil {
			// Set default MaxSurge as 25% by default.
			maxSurge := intstr.FromString("25%")
			strategy.RollingUpdate.MaxSurge = &maxSurge
		}
```

Reply:
In [API Overview](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#rollingupdatedaemonset-v1-apps) it is stated clearly. 
```
This can not be 0 if MaxUnavailable is 0. Absolute number is calculated from percentage by rounding up to a minimum of 1. Default value is 0. Example: when this is set to 30%, at most 30% of the total number of nodes that should be running the daemon pod (i.e. status.desiredNumberScheduled) can have their a new pod created before the old pod is marked as deleted.
```